lib/active_model/serializer.rb

```diff
@@ -80,24 +80,42 @@ def self.serializers_cache
 
     # @api private
     # Find a serializer from a class and caches the lookup.
-    # Preferentially returns:
-    #   1. class name appended with "Serializer"
-    #   2. try again with superclass, if present
-    #   3. nil
     def self.get_serializer_for(klass)
       return nil unless config.serializer_lookup_enabled
       serializers_cache.fetch_or_store(klass) do
-        # NOTE(beauby): When we drop 1.9.3 support we can lazify the map for perfs.
-        serializer_class = serializer_lookup_chain_for(klass).map(&:safe_constantize).find { |x| x && x < ActiveModel::Serializer }
-
-        if serializer_class
-          serializer_class
-        elsif klass.superclass
-          get_serializer_for(klass.superclass)
-        end
+        get_serializer_class(klass)
+      end
+    end
+
+    # @api private
+    # Preferentially returns:
+    #   1. class name appended with "Serializer"
+    #   2. try again with superclass, if present
+    #   3. (optional) runs `on_serializer_not_found` policy if serializer is not found
+    #   4. nil
+    def self.get_serializer_class(klass)
+      serializer_class = find_serializer_class(klass)
+
+      if serializer_class
+        serializer_class
+      elsif (ancestor_serializer_classes = ancestor_serializer_classes(klass)).present?
+        ancestor_serializer_classes.first
+      elsif config.on_serializer_not_found.respond_to?(:call) && klass.respond_to?(:serialized_attributes)
```

Comment:
1. The config setup you mentioned is probably the best way to go. However, if we setup a default strategy that returns a symbol, we will brake backwards compatibility for anyone relying on the historical behaviour of returning `nil`. 

2. You're right about checking for callability. I think we can actually do `elsif config.on_serializer_not_found` - if it's an empty proc it will run, and when someone messes up the config value (ie setting it to `:no_serializer_found` instead of `-> { :no_serializer_found }` he will get an explicit error about what's wrong with the code. WDYT?

2. For [`:serialized_attributes`](http://apidock.com/rails/ActiveRecord/Base/serialized_attributes/class), I'm checking whether this is actually something that represents a serializable AR resource. I'm aware that this excludes non AR-based objects. Maybe change this to `klass.respond_to?(:serialized_attributes) || klass.respond_to?(:serialized_hash)`? This will handle both cases. I would much rather not couple this to AR via `serialized_attributes` but I could not find any other way of detecting is the object is both AR-based and serializable.